### PR TITLE
gitignore `Loop/DerivedAssets.xcassets` & `WatchApp/DerivedAsserts.xcassets`

### PR DESCRIPTION
**Pull request by rickpasetto**
This adds `Loop/DerivedAssets.xcassets` & `WatchApp/DerivedAsserts.xcassets` to `.gitignore`, but _leaves_ `Contents.json`, which is unmodified by the build script, but is required so that Xcode doesn't give an error in building.  Also moves the `build-derived-assets.sh` script to a pre-build step, which might be a better place for it.

LW PR: https://github.com/tidepool-org/LoopWorkspace/pull/84

### CODE REVIEW

**Review by darinkrauss — APPROVED** [review]
Approving, but I'm not providing any warranty that this won't break something downstream.

File: Cartfile.resolved [hunk 1]
```diff
@@ -1,8 +1,8 @@
 github "LoopKit/Amplitude-iOS" "2137d5fd44bf630ed33e1e72d7af6d8f8612f270"
```
Comment (darinkrauss) [inline Cartfile.resolved]: Since the `Cartfile` and `Cartfile.resolved` reference the DIY repos (i.e. `LoopKit` prefix), this file should not be updated unless you actually pulled in changes from the DIY repo. And, if you did that then the DIY pull should be a completely separate PR explicitly for pulling in DIY changes.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
I don't think the `Cartfile.resolved` changes should be in the PR.

**Comment by rickpasetto** [comment]
You're right.  I'll back it out.

**Review by darinkrauss — APPROVED** [review]
LGTM! 🍡
